### ISSUE TITLE
Previews in backend

### ISSUE THREAD

**Issue by Boorj** [bolt/bolt#4474]
enchancement suggestion:
I suggest change preview behaviour a little bit in backend. 
For image fieldtype bolt fills preview-viewport (200×150), it equals to css's `background-size: cover`.
Sometimes it's good, sometimes - not.
![2015-11-17 00-39-39](https://cloud.githubusercontent.com/assets/2145313/11191159/fae60c8a-8cc3-11e5-8673-f5fc5feeacef.png)  ![2015-11-17 00-39-09](https://cloud.githubusercontent.com/assets/2145313/11191165/01af3f8c-8cc4-11e5-8e3c-a18fd5e03d8c.png)
(the original one was 
![45](https://cloud.githubusercontent.com/assets/2145313/11191182/15679c36-8cc4-11e5-87ea-b41fbce76c86.png))

So there are few ways to improve that:
1. make all pictures to be fit to that viewport by changing fit method from crop (`thumbs/200x150c`) to resize (`thumbs/200x150r`)
2. other way - to make a resized (not cropped) picture and then set `background-size: contain`
3. nice solution: make viewport contain picture (fit-mode) and on hover - make it fill (fill mode, bg-size: cover). it could work even on touch screens on tap.
4. nicest solution: make it contain by default, and cover on hover, and when you move mouse through viewport - it scrolls zoomed image inside viewport, according to mouse coordinates. But that takes a portion of js.

**Comment by Boorj** [comment]
#### Comparsion
```html
<div class="content-preview imageholder">
    <img alt="Preview" height="150" src="/thumbs/200x150c/residentials/12.png" width="200">
</div>
```
![2015-11-17 01-14-44](https://cloud.githubusercontent.com/assets/2145313/11192009/d042bbfe-8cc8-11e5-975b-db6ec1556558.png)
**vs.**
```html
<div class="content-preview imageholder" 
       style="height: 150px;
                 width: 200px;
                 background: url(/thumbs/1200x150r/residentials/12.png) white no-repeat 50% 50% /contain">
</div>
```
![2015-11-17 01-14-27](https://cloud.githubusercontent.com/assets/2145313/11192017/d9ee0bc2-8cc8-11e5-99fd-b6ef50f28b0e.png)

Of course, all css attributes except image url could be moved to .css file. And also some `:hover` rulesets could be added (like background-size: cover) as i said before.
Background-size property is css3, but it could work for older versions as it's being shown now, if you set image size correctly.

**Comment by rarila** [comment]
Making the preview a widget is very high on my list. As soon as this is done doing JS vodoo on it is a breeze.

**Comment by Boorj** [comment]
Please don't forget about 
![2015-11-17 00-31-15](https://cloud.githubusercontent.com/assets/2145313/11205754/58910cfc-8d34-11e5-8a2e-6896ca7fe914.png)

:leaves:

**Comment by SahAssar** [comment]
@rarila Is this easier (or maybe fixed?) with the buic model?

**Comment by rarila** [comment]
Just a spare time problem… But I think one of 1) to 3) will do as there's correct working fullscreen preview on click again

**Comment by bobdenotter** [comment]
Fix for the first part incoming. 

Marking this as "wont fix" for the listing views. In _most_ cases, the current cropping is preferred: If you have photos, it'll crop all of them to the same size, giving a consistent look. If we switched to cropping them while keeping aspect ratio, that page looks like a mess.

**Comment by Boorj** [comment]
Actually, i can see no problem to make it with 'resize' option and do everything with css:
```css
.preview           { background-size: cover;}
.preview:hover { background-size: contain;}
```

**Comment by bobdenotter** [comment]
I quickly tried it, and it gave a weird, juttering effect. If we're going to work onthis, we should first make a plan of what the UX/UI of these things is going to be like. Perhaps we could make a nice Vue component of it. ;-)

**Comment by GawainLynch** [comment]
>  If we're going to work onthis, we should first make a plan of what the UX/UI of these things is going to be like. 

That would be awesome if we could get some planning around this :+1: 

> Perhaps we could make a nice Vue component of it. ;-)

:+1: :+1: :+1:

**Comment by bobdenotter** [comment]
Fixed (for now) by #5847
